Wes test

WesleyAnderson349 opened the pull request:
Lots of changes 
Data Import 
Fixes for Linking Responses to Users 
Changes To Avoid Entity not Found exceptions that can happen when pulling Homework Questions by Protocol ID/Protocol Template IDs

WesleyAnderson349 committed:
Small changes and fixes

WesleyAnderson349 committed:
Homework Question changes

WesleyAnderson349 committed:
Fixes for the Link Response

WesleyAnderson349 committed:
Safety commit

WesleyAnderson349 committed:
Data Import changes

WesleyAnderson349 committed:
Data Import fixes

Some small changes Data Import + changes to get All homework questions to handle questions without Category ID's more gracefully (needed for legacy data and some of the imports lacked Categories)

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/dto/importHomeworkQuestionDto.java:
```diff
@@ -0,0 +1,30 @@
+  private QuestionTypeEnum questionType; // Homework Question Type Column G. 6
+  private Boolean isRequired; // is Required Column H.  7
+  private String expectedResponse; // Expected Response (comma Seperated list) Column I. 8
+  //private ExpectedResponse triggeringResponse; // Triggering Response
```
Let's remove the commented code and comments here

timothy-stratton-cairn commented on src/main/resources/db/changelog/2024-11-18-drop-protocol-id-from-response.yaml:
```diff
@@ -0,0 +1,12 @@
+      onFail: HALT
+      onError: HALT
+  - changeSet:
+      id: add-user-id-to-homework-response
```
The `id` here isn't reflective of the work being done, I think.

WesleyAnderson349 committed:
Update importHomeworkQuestionDto.java

WesleyAnderson349 committed:
corrected

timothy-stratton-cairn commented on Waypoints_Data_Import_Template (1).xlsx:
We can remove this from the repository. Not best to keep data of this sort in source control

WesleyAnderson349 committed:
GetProtocolByIdEndpoints Changes

Changes to pull User Ids with the protocols

WesleyAnderson349 committed:
Protocol Changes + Question Response filter

Changes to Add Protocol to instead validate and use a User Id instead of Household
Added the filter Logic to GetQuestionAndResponsePairs endpoint to only show the most recent response

WesleyAnderson349 committed:
Fixes for protocol Assigned to Userr

WesleyAnderson349 committed:
Code Style Changes

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/protocol/GetAllProtocolsByUserIdEndpoint.java:
```diff
@@ -45,9 +59,83 @@ public GetAllProtocolsByUserIdEndpoint(ProtocolDataService protocolDataService)
-    List<Protocol> protocols = protocolDataService.findByUserId(userId);
-    return ResponseEntity.ok(protocols);
+    return ResponseEntity.ok(
+        AccountProtocolListDto.builder()
```
This should become a mapper. This is a bit verbose and could be error prone if copied to other places

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/entity/Protocol.java:
```diff
@@ -77,7 +78,7 @@ public class Protocol extends BaseEntity {
   @JoinColumn(name = "parent_protocol_id")
   @OneToMany(fetch = FetchType.EAGER, cascade = {CascadeType.MERGE, CascadeType.PERSIST})
-  private Set<ProtocolStep> protocolSteps;
+  private Set<ProtocolStep> protocolSteps = new HashSet<>();
```
Is there a particular need to have this on the entity as opposed to having it instantiated and set by a caller? I would suggest that we pull this into the caller instead of setting this at the level. It's quite a low level. Moreover, this is an entity and the values should be set by our ORM framework. The instantiation here breaks the standard.

WesleyAnderson349 replied:
was something I tried to fix a bug I was running into, wasn't necessary it's been removed.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/entity/ProtocolTemplateLinkedStepTemplate.java:
```diff
@@ -29,6 +29,7 @@ public class ProtocolTemplateLinkedStepTemplate extends BaseEntity {
   private ProtocolTemplate protocolTemplate;
 
   @JoinColumn(name = "step_template_id", nullable = false)
+
```
Can remove this space

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/data/QuestionResponsePairDataService.java:
```diff
@@ -31,6 +30,10 @@ public QuestionResponsePairDataService(HomeworkQuestionDataService questionServi
     this.mapper = mapper;
   }
 
+  /*public QuestionResponsePairListDto findALLQuestionResponsePairsByProtocolId(Long protocolId) {
```
Can we remove this commented code?

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/helper/QuestionMapper.java:
```diff
@@ -0,0 +1,27 @@
+  @Mapping(source = "isRequired", target = "required")
+  @Mapping(source = "category", target = "category")
+  @Mapping(source = "triggersProtocolCreation", target = "triggersProtocolCreation")
+  //@Mapping(source = "triggeringResponse", target = "triggeringResponse")
```
And this commented code

timothy-stratton-cairn commented on src/main/resources/db/changelog/2024-05-18-add-protocol-statuses-to-system.yaml:
```diff
@@ -28,7 +28,7 @@ databaseChangeLog:
       changes:
         - sql:
             dbms: 'h2, mysql'
-            sql: ALTER TABLE protocol 
```
Here you're going to face issues with the checksum. Can we revert these files

timothy-stratton-cairn commented on src/main/resources/db/changelog/2024-10-16-create-category-table.yaml:
```diff
@@ -9,13 +9,13 @@ databaseChangeLog:
-                  UNIQUE KEY (`name`)
-                  )ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
\ No newline at end of file
+            `id` bigint NOT NULL AUTO_INCREMENT,
```
We should exclude all yaml files from your reformatter

WesleyAnderson349 committed:
PR Changes

WesleyAnderson349 committed:
Mapper changes
